src/libs/EmojiUtils.js

```diff
@@ -195,11 +196,20 @@ function addToFrequentlyUsedEmojis(newEmoji) {
         }))
         .value();
 
-    // Sort the list and take the first maxFrequentEmojiCount items
-    const frequentEmojiListOrdered = lodashOrderBy(mergedEmojisWithFrequent, ['count', 'lastUpdatedAt'], ['desc', 'desc'])
-        .slice(0, maxFrequentEmojiCount);
+    const maxFrequentEmojiCount = (CONST.EMOJI_FREQUENT_ROW_COUNT * CONST.EMOJI_NUM_PER_ROW) - uniqueEmojisWithCount.length;
 
-    User.updateFrequentlyUsedEmojis(frequentEmojiListOrdered);
+    // After we get the new count and lastUpdatedAt, take out the updated unique emojis from mergedEmojisWithFrequent
+    const emojisWithNewCount = lodashIntersectionBy(mergedEmojisWithFrequent, uniqueEmojisWithCount, 'name');
+    const frequentEmojisList = lodashDifferenceBy(mergedEmojisWithFrequent, emojisWithNewCount, 'name');
```

Comment:
Actually, `frequentEmojisList` contains frequently used emojis list - the new emoji we insert.

From the example,
`mergedEmojisWithFrequent intersect uniqueEmojisWithCount` = [1,2,7,8]
basically the intersection will always the same emoji with `uniqueEmojisWithCount`, but we want the emoji array from `mergedEmojisWithFrequent` because it contains the updated count.